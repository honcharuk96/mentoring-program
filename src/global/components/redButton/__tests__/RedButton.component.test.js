import React from 'react';
import { render } from '@testing-library/react';
import RedButtonComponent from '../redButton.component';
import renderer from 'react-test-renderer'
import 'jest-styled-components';

describe('RedButtonComponent', function () {
    it('should have text "test"', function () {
        let { getByText } = render(<RedButtonComponent  text={'test'}/>);
        expect(getByText('test'));
    });
    it('default button', () => {
        const tree = renderer.create(<RedButtonComponent  text={'test'}/>).toJSON()
        expect(tree).toMatchSnapshot()
        expect(tree).toHaveStyleRule('background', '#f65261')
        expect(tree).toHaveStyleRule('color', '#ffffff')
    })
    it('revert button', () => {
        const tree = renderer.create(<RedButtonComponent revertColor text={'test'}/>).toJSON()
        expect(tree).toMatchSnapshot()
        expect(tree).toHaveStyleRule('background', '#232323')
        expect(tree).toHaveStyleRule('color', '#f65261')
    })
});
