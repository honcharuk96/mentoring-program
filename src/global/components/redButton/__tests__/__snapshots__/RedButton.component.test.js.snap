// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedButtonComponent default button 1`] = `
.c0 {
  padding: 18px 70px;
  border-radius: 4px;
  background: #f65261;
  color: #ffffff;
  border: 1px solid #f65261;
  font-size: 20px;
  font-style: normal;
  font-weight: 500;
  line-height: 24px;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
}

<div
  className="c0"
  onClick={null}
>
  test
</div>
`;

exports[`RedButtonComponent revert button 1`] = `
.c0 {
  padding: 18px 70px;
  border-radius: 4px;
  background: #232323;
  color: #f65261;
  border: 1px solid #f65261;
  font-size: 20px;
  font-style: normal;
  font-weight: 500;
  line-height: 24px;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
}

<div
  className="c0"
  onClick={null}
>
  test
</div>
`;
